test/odbc/test_pg_datatype_float.cpp

```diff
@@ -0,0 +1,610 @@
+        std::cout<<"insert value is "<<col_results[j]<<'\n';
+        ASSERT_EQ(col_results[j], StringToDouble(inserted_values[i][j]));
+        // std::cout<<"length is "<<(col_len[j])<<'\n';
+        ASSERT_EQ(col_len[j], BYTES_EXPECTED);
+      
+      } 
+      else 
+        ASSERT_EQ(col_len[j], SQL_NULL_DATA);
+    }
+  }
+
+  // Assert that there is no more data
+  rcode = SQLFetch(odbcHandler.GetStatementHandle());
+  ASSERT_EQ(rcode, SQL_NO_DATA);
+
+  odbcHandler.CloseStmt();
+  odbcHandler.ExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+}
+
+TEST_F(PSQL_DataTypes_Float, Insertion_Failure) {
+
+
+  SQLLEN col_len[NUM_COLS];
+
+  RETCODE rcode;
+  OdbcHandler odbcHandler; 
+
+  vector<vector<string>> inserted_values = {
+    {"0","1e309","0"},
+    {"1", "0","-1e309"},
+    {"2", "1e-325","0"},
+    {"3", "0","-1e-324"}
+     // 309 would succeed the failure   range  -1e308 to -1e-307  0 and 0 1e307 to 1e308     
+  };
+
+  // Create table
+  odbcHandler.ConnectAndExecQuery(CreateTableStatement(TABLE_NAME, TABLE_COLUMNS));
+  odbcHandler.CloseStmt();
+
+  // Insert invalid values in table and assert error
+  for (int i = 0; i < inserted_values.size(); i++) {
+
+    string insert_string = "(";
+    string comma{};
+
+    // create insert_string (1, ..., ..., ...)
+    for (int j = 0; j < NUM_COLS; j++) {
+      insert_string += comma + inserted_values[i][j];
+      comma = ",";
+    }
+    insert_string += ")";
+    // std::cout<<"Current Value is: "<< insert_string;
+    rcode = SQLExecDirect(odbcHandler.GetStatementHandle(), (SQLCHAR*) InsertStatement(TABLE_NAME, insert_string).c_str(), SQL_NTS);
+    ASSERT_EQ(rcode, SQL_ERROR);
+    odbcHandler.CloseStmt();
+  }
+
+  // Select all from the table to make sure nothing was inserted
+  odbcHandler.ExecQuery(SelectStatement(TABLE_NAME, {"*"}, vector<string> {COL_NAMES[0]}));
+  rcode = SQLFetch(odbcHandler.GetStatementHandle());
+  ASSERT_EQ(rcode, SQL_NO_DATA);
+
+  odbcHandler.CloseStmt();
+  odbcHandler.ExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+}
+
+TEST_F(PSQL_DataTypes_Float, Update_Success) {
+
+  const int BUFFER_LENGTH = 8192;
+  const int AFFECTED_ROWS_EXPECTED = 1;
+  const int BYTES_EXPECTED = 8;
+  const string PK_VAL = "1";
+
+  double col_results[NUM_COLS];
+  SQLLEN col_len[NUM_COLS];
+  SQLLEN affected_rows;
+
+  RETCODE rcode;
+  OdbcHandler odbcHandler;
+
+  vector<vector<string>> inserted_values = {
+    {PK_VAL, "0.1", "0.2"} 
+  };
+
+  vector<vector<string>> updated_values = {
+    // standard values
+    {PK_VAL, "1e50", "-1e50"}, // max values
+    {PK_VAL, "0", "0"},
+    {PK_VAL,float_15,float_383} // min values
+    
+  };
+
+  vector<tuple<int, int, SQLPOINTER, int, SQLLEN* >> bind_columns{};
+
+  // initialize bind_columns
+  for (int i = 0; i < NUM_COLS; i++) {
+    tuple<int, int, SQLPOINTER, int, SQLLEN*> tuple_to_insert(i+1, SQL_C_DOUBLE, (SQLPOINTER) &col_results[i], BUFFER_LENGTH, &col_len[i]);
```

Comment:
```suggestion
    tuple<int, int, SQLPOINTER, int, SQLLEN*> tuple_to_insert(i + 1, SQL_C_FLOAT, (SQLPOINTER) &col_results[i], BUFFER_LENGTH, &col_len[i]);
```
&& `SQL_C_DOUBLE` -> `SQL_C_FLOAT`